modules/@apostrophecms/modal/ui/apos/components/AposModal.vue

```diff
@@ -232,47 +251,32 @@ export default {
       this.$emit('esc');
     },
     trapFocus () {
-      // Adapted from https://uxdesign.cc/how-to-trap-focus-inside-modal-to-make-it-ada-compliant-6a50f9a70700
-      // All the elements inside modal which you want to make focusable.
-      const focusableElements = [
-        'button',
+      const elementSelectors = [
+        '[tabindex]',
         '[href]',
         'input',
         'select',
         'textarea',
-        '[tabindex]:not([tabindex="-1"]'
+        'button'
       ];
-      const focusableString = focusableElements.join(', ');
-      const modalEl = this.$refs.modalEl;
-      const focusables = modalEl.querySelectorAll(focusableString);
-      const firstFocusableElement = focusables[0];
-      const lastFocusableElement = focusables[focusables.length - 1];
 
-      modalEl.addEventListener('keydown', cycleFocusables);
+      const selector = elementSelectors
+        .map(addExcludingAttributes)
+        .join(', ');
 
-      firstFocusableElement.focus();
+      this.elementsToFocus = [ ...this.$refs.modalEl.querySelectorAll(selector) ]
+        .filter(isElementVisible);
+      console.log('🚀 ~ file: AposModal.vue:269 ~ trapFocus ~ this.focusedElement:', this.focusedElement);
+      console.log('🚀 ~ file: AposModal.vue:269 ~ trapFocus ~ this.elementsToFocus:', this.elementsToFocus);
 
-      function cycleFocusables (e) {
-        const isTabPressed = e.key === 'Tab' || e.code === 'Tab';
-        if (!isTabPressed) {
-          return;
-        }
+      // Focus focusedElement in priority to keep it selected after a refresh:
+      (this.focusedElement || this.elementsToFocus[0]).focus();
 
-        if (e.shiftKey) {
-          // If shift key pressed for shift + tab combination
-          if (document.activeElement === firstFocusableElement) {
-            // Add focus for the last focusable element
-            lastFocusableElement.focus();
-            e.preventDefault();
-          }
-        } else {
-          // If tab key is pressed
-          if (document.activeElement === lastFocusableElement) {
-            // Add focus for the first focusable element
-            firstFocusableElement.focus();
-            e.preventDefault();
-          }
-        }
+      function addExcludingAttributes(element) {
+        return `${element}:not([tabindex="-1"]):not([disabled]):not([type="hidden"]):not([aria-hidden])`;
```

Comment:
this is important, avoid focusing elements that are hidden/disabled or that have a negative tabindex